tests/components/konnected/test_config_flow.py

```diff
@@ -0,0 +1,677 @@
+
+    # zone 4
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_switch"
+
+    # zone 8
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            "name": "switcher",
+            "activation": "low",
+            "momentary": 50,
+            "pause": 100,
+            "repeat": 4,
+        },
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_switch"
+
+    # zone out1
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_switch"
+
+    # zone alarm1
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={}
+    )
+    assert result["type"] == "create_entry"
+    assert result["data"] == {
+        "host": "1.2.3.4",
+        "port": 1234,
+        "id": "112233445566",
+        "blink": True,
+        "discovery": True,
+        "binary_sensors": [
+            {"zone": "2", "type": "door", "inverse": False},
+            {"zone": "6", "type": "window", "name": "winder", "inverse": True},
+            {"zone": "10", "type": "door", "inverse": False},
+        ],
+        "sensors": [
+            {"zone": "3", "type": "dht", "poll_interval": 3},
+            {"zone": "7", "type": "ds18b20", "name": "temper", "poll_interval": 3},
+            {"zone": "11", "type": "dht", "poll_interval": 3},
+        ],
+        "switches": [
+            {"activation": "high", "zone": "4"},
+            {
+                "zone": "8",
+                "name": "switcher",
+                "activation": "low",
+                "momentary": 50,
+                "pause": 100,
+                "repeat": 4,
+            },
+            {"activation": "high", "zone": "out1"},
+            {"activation": "high", "zone": "alarm1"},
+        ],
+    }
+
+
+async def test_ssdp(hass, mock_panel):
+    """Test a panel being discovered."""
+    mock_panel.get_status.return_value = {
+        "mac": "11:22:33:44:55:66",
+        "name": "Konnected",
+    }
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": "ssdp"},
+        data={
+            "ssdp_location": "http://1.2.3.4:1234/Device.xml",
+            "manufacturer": config_flow.KONN_MANUFACTURER,
+            "modelName": config_flow.KONN_MODEL,
+        },
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "io"
+    assert result["description_placeholders"]["model"] == "Konnected Alarm Panel"
+
+    # ensure the discovered info is saved
+    # pylint: disable=protected-access
+    flow = hass.config_entries.flow._progress[result["flow_id"]]
+    assert flow.device_id == "112233445566"
```

Comment:
Don't assert these things. These are implementation details that may change later. We should only assert the flow results.

Reply:
Understood - I'll fix these in the upcoming options conversion commit.